Pod/Classes/LocationManager.swift

```diff
@@ -10,244 +10,278 @@
+                            reject(LocationManagerError.cannotFetchLocation)
                         }
                     }
                 }
             }
         }
     }
     
-    internal func locationRequestDidTimeout(request: LocationRequest) {
-        self.removeLocationRequest(request)
+    internal func locationRequestDidTimeout(_ request: LocationRequest) {
+        removeLocationRequest(request)
     }
     
-    internal func removeLocationRequest(request: LocationRequest) {
-        if let index = self.locationRequests.indexOf(request) {
-            self.locationRequests.removeAtIndex(index)
+    internal func removeLocationRequest(_ request: LocationRequest) {
+
+        if let index = locationRequests.index(of: request) {
+            locationRequests.remove(at: index)
         }
     }
     
-    // MARK: - location observers
+    // MARK: - Location observers
     
-    public class func addLocationObserver(observer: LocationObserver, desiredAccuracy: CLLocationAccuracy? = nil, distanceFilter: CLLocationDistance? = nil, minimumTimeInterval: NSTimeInterval? = nil, maximumTimeInterval: NSTimeInterval? = nil) {
-        self.sharedManager.addLocationObserver(observer, desiredAccuracy: desiredAccuracy, distanceFilter: distanceFilter, minimumTimeInterval: minimumTimeInterval, maximumTimeInterval: maximumTimeInterval)
+    open class func addLocationObserver(_ observer: LocationObserver, desiredAccuracy: CLLocationAccuracy? = nil, distanceFilter: CLLocationDistance? = nil, minimumTimeInterval: TimeInterval? = nil, maximumTimeInterval: TimeInterval? = nil) {
+        sharedManager.addLocationObserver(observer, desiredAccuracy: desiredAccuracy, distanceFilter: distanceFilter, minimumTimeInterval: minimumTimeInterval, maximumTimeInterval: maximumTimeInterval)
     }
     
-    public func addLocationObserver(observer: LocationObserver, desiredAccuracy: CLLocationAccuracy? = nil, distanceFilter: CLLocationDistance? = nil, minimumTimeInterval: NSTimeInterval? = nil, maximumTimeInterval: NSTimeInterval? = nil) {
+    open func addLocationObserver(_ observer: LocationObserver, desiredAccuracy: CLLocationAccuracy? = nil, distanceFilter: CLLocationDistance? = nil, minimumTimeInterval: TimeInterval? = nil, maximumTimeInterval: TimeInterval? = nil) {
+
         let item = LocationObserverItem(locationObserver: observer, locationManager: self, desiredAccuracy: desiredAccuracy, distanceFilter: distanceFilter, minimumTimeInterval: minimumTimeInterval, maximumTimeInterval: maximumTimeInterval)
-        self.locationObservers.insert(item)
+        locationObservers.insert(item)
         
-        self.startUpdatingLocationIfNeeded()
+        startUpdatingLocationIfNeeded()
     }
     
-    public class func removeLocationObserver(observer: LocationObserver) {
-        self.sharedManager.removeLocationObserver(observer)
+    open class func removeLocationObserver(_ observer: LocationObserver) {
```

Comment:
I would also refactor naming according to Swift 3 naming guidelines. For example this method should be 
```swift
open class func remove(locationObserver: LocationObserver) {
}
```